mapgen/iwyu-mapgen-macos.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python3
+
+##===--- iwyu-mapgen-macos.py ---------------------------------------------===##
+#
+#                     The LLVM Compiler Infrastructure
+#
+# This file is distributed under the University of Illinois Open Source
+# License. See LICENSE.TXT for details.
+#
+##===----------------------------------------------------------------------===##
+
+""" Generate mappings for macOS libc.
+
+Map headers that either start with an underscore or are architecture-specific.
+"""
+import argparse
+import json
+import sys
+
+from operator import itemgetter
+from pathlib import Path
+
+
+PATTERN_MAPPINGS = {
+    'pthread_*': 'pthread.h',
+    'sys/_types/_fd_*': 'sys/select.h',
+    'sys/_types/_in_*': 'netinet/in.h',
+    'sys/_types/_o_*': 'fcntl.h',
+    'sys/_types/_s_*': 'sys/stat.h',
+    'sys/_types/_seek_*': 'stdio.h',
```

Comment:
We can get rid of all of these special cases by adding more symbols to iwyu_include_picker.cc, since they're all POSIX symbols with standard locations. The benefit of that is that it would also work on other systems. See #1706.

Reply:
I wonder if it makes sense to move ahead with these independently, and merge this first?

I'd like to see some test output from #1706, and that's currently blocked by #1700.

Reply:
Sure, sounds good to me.

Reply:
I'll try to get to it tomorrow.